Ray.jl/src/object_store.jl

```diff
@@ -20,7 +20,7 @@ if run in an `@async` task.
 If the task that generated the `ObjectID` failed with a Julia exception, the
 captured exception will be thrown on `get`.
 """
-get(oid::ray_jll.ObjectIDAllocated) = _get(take!(ray_jll.get(oid)))
+get(oid::ray_jll.ObjectIDAllocated; timeout_ms::Int=-1) = _get(take!(ray_jll.get(oid, timeout_ms)))
```

Comment:
We need to allow passing in this keyword for each `get` method otherwise users will be unable to write generic functions. 

I'll also mention we're deviating from the Python interface which accepts the timeout in seconds: https://github.com/ray-project/ray/blob/10c461c6d03b11bfa3f03a1a95907b6db27aac7d/python/ray/_private/worker.py#L754. I can understand why just pass in milliseconds as this avoids any floating point issues but since Julia functions typically take seconds for timeouts it is probably best to use seconds here again.

Overall, for the Julia interface I think it would be better to have an `isready` and blocking `get`.

Reply:
I agree that separating isready and get/fetch is reasonable, but IMO the current situation is not acceptable: right now, `get` on an un-ready future can hard lock the julia session in such a way that the only option is either to kill it at teh OS level or stop the ray server.

Reply:
that is, the blocking is not the issue _per se_, rather that teh blocking happens at a level that we have no control over on teh julia side.  even in an `@async` block, for instance, it'll block indefinitely

Reply:
So is the plan to utilize a small internal timeout here until we reach the user's specified timeout? 

What I want to avoid is providing a high-level user interface which we don't want to support in the long term. Did you know about the [`Contains`](https://github.com/ray-project/ray/blob/a03efd9931128d387649dd48b0e4864b43d3bfb4/src/ray/core_worker/core_worker.h#L683-L691) core worker function? We could possibly utilize that function to perform a similar kind of check on the Julia side (this is probably how we'd implement `isready`). I'll note I haven't had to use this function myself so we'll have to validate its behavior.

Reply:
> So is the plan to utilize a small internal timeout here until we reach the user's specified timeout?

basically yes, unless we can find some way to do the long-poll get on a separate thread or something.

> What I want to avoid is providing a high-level user interface which we don't want to support in the long term. 

Yup, we're on the same page there.  I think we can have a separation between the `ray_jll.ObjectID` and `Ray.ObjectRef` methods: the objectrefs won't have a manual timeout but the ObjectIDs would...

> Did you know about the [Contains](https://github.com/ray-project/ray/blob/a03efd9931128d387649dd48b0e4864b43d3bfb4/src/ray/core_worker/core_worker.h#L683-L691) core worker function? We could possibly utilize that function to perform a similar kind of check on the Julia side (this is probably how we'd implement isready). I'll note I haven't had to use this function myself so we'll have to validate its behavior.

yeah I was just looking at contains, I think that's what we'll want to use.  Will probably want to use a v. short timeout there too to make things async friendly, and so will still need to find some way to handle timeouts.